nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/ConsumeJMS.java

```diff
@@ -86,6 +86,31 @@
             .defaultValue(CLIENT_ACK.getValue())
             .build();
 
+    static final PropertyDescriptor DURABLE_SUBSCRIBER = new PropertyDescriptor.Builder()
+                                                             .name("Durable subscription")
+                                                             .description("If destination is Topic if present then make it the consumer durable. " +
+                                                                          "@see https://docs.oracle.com/javaee/7/api/javax/jms/Session.html#createDurableConsumer-javax.jms.Topic-java.lang.String-")
+                                                             .required(false)
+                                                             .expressionLanguageSupported(true)
+                                                             .defaultValue("true")
+                                                             .allowableValues("true", "false")
+                                                             .build();
+    static final PropertyDescriptor SHARED_SUBSCRIBER = new PropertyDescriptor.Builder()
+                                                            .name("Shared subscription")
+                                                            .description("If destination is Topic if present then make it the consumer shared. " +
+                                                                         "@see https://docs.oracle.com/javaee/7/api/javax/jms/Session.html#createSharedConsumer-javax.jms.Topic-java.lang.String-")
+                                                            .required(false)
+                                                            .expressionLanguageSupported(true)
+                                                            .defaultValue("false")
+                                                            .allowableValues("true", "false")
+                                                            .build();
+    static final PropertyDescriptor SUBSCRIPTION_NAME = new PropertyDescriptor.Builder()
+                                                            .name("Subscription Name")
+                                                            .description("The name of the subscription to use if destination is Topic and is shared or durable.")
+                                                            .required(false)
+                                                            .expressionLanguageSupported(true)
+                                                            .build();
```

Comment:
If a property descriptor doesn't have any validator, it just becomes invalid. Please add NON_EMPTY_VALIDATOR.

```
.addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

![image](https://user-images.githubusercontent.com/1107620/27617494-a5362bfe-5bf1-11e7-9b95-9ded43bdd6b6.png)

I added above code temporarily to proceed testing.